src/tiled/tilesetdock.cpp

```diff
@@ -758,20 +758,20 @@ void TilesetDock::exportTileset()
     if (!tileset)
         return;
 
-    const QLatin1String extension(".tsx");
-    QString suggestedFileName = QFileInfo(mMapDocument->fileName()).path();
-    suggestedFileName += QLatin1Char('/');
-    suggestedFileName += tileset->name();
-    if (!suggestedFileName.endsWith(extension))
-        suggestedFileName.append(extension);
```

Comment:
Please don't just rip out the suggested file name logic. You'll need to combine the stored path with the suggested file name.